WikiDrips/WikiRequest.swift

```diff
@@ -7,17 +7,67 @@
 //
 
 import Foundation
-import Accounts
-import Social
-import CoreLocation
 
 public class WikiRequest
 {
-    public func fetchWikiDoc(handler: @escaping ([WikiDoc]) -> Void) {
-//      TODO: make api requests
-            var wikiDocs = [WikiDoc]()
-            var wikiDocArray: NSArray?
-        
-
+    let session = URLSession.shared
+    let baseEndpoint: String = "http://en.wikipedia.org/w/api.php?action=query&list=search&srwhat=text&sroffset=0&format=json"
+    var wikiDocs = [[WikiDoc]]()
+    let searchText: String
+    
+    init?(searchText: String) {
+        self.searchText = searchText
+    }
+    
+    public func fetchWikiDocs(handler: @escaping ([[WikiDoc]]) -> Void) {
+        guard let urlRequest = urlRequest(searchText: searchText) else {
+            return
+        }
+        let task = session.dataTask(with: urlRequest) {
+            (data, response, error) in
+            // check for any errors
+            guard error == nil else {
+                print("error calling GET on \(self.baseEndpoint)")
+                print(error!)
+                return
+            }
+            // make sure we got data
+            guard let data = data else {
+                print("Error: did not receive data")
+                return
+            }
+            // parse the result as JSON
+            do {
+                guard let response = try? JSONSerialization.jsonObject(with: data, options: []) else {
+                    print("error trying to convert data to JSON")
+                    return
+                }
+                if let dictionary = response as? [String: Any] {
```

Comment:
Use `guards` instead of [Pyramid of Doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming))